core-library/src/main/java/org/silverpeas/core/security/authorization/ComponentInstanceSimpleDocumentAccessControlExtension.java

```diff
@@ -0,0 +1,108 @@
+ * {@link javax.inject.Named} annotation by a name satisfying the following convention
+ * <code>[COMPONENT NAME]InstanceSimpleDocumentAccessControlExtension</code>. For example, for an
+ * application Kmelia, the implementation must be qualified with <code>@Named
+ * ("kmeliaInstanceSimpleDocumentAccessControlExtension")
+ * </code>
+ * <p>
+ * @author silveryocha
+ */
+public interface ComponentInstanceSimpleDocumentAccessControlExtension {
+
+  /**
+   * Constants are predefined value used by a contribution manager to work with and that carries a
+   * semantic that has to be shared by all the implementations of this interface.
+   */
+  class Constants {
+
+    private Constants() {}
+
+    /**
+     * The predefined suffix that must compound the name of each implementation of this interface.
+     * An implementation of this interface by a Silverpeas application named Kmelia must be named
+     * <code>kmelia[NAME_SUFFIX]</code> where NAME_SUFFIX is the predefined suffix as defined below.
+     */
+    public static final String NAME_SUFFIX = "InstanceSimpleDocumentAccessControlExtension";
+  }
+
+  /**
+   * Gets the {@link ComponentInstanceSimpleDocumentAccessControlExtension} according to the given
+   * identifier of component instance.
+   * <p>
+   * Instances of {@link ComponentInstanceSimpleDocumentAccessControlExtension} are request scoped
+   * (or thread scoped on backend treatments).
+   * </p>
+   * @param instanceId the identifier of a component instance from which the qualified name of the
+   * implementation will be extracted.
+   * @return a {@link ComponentInstanceSimpleDocumentAccessControlExtension} implementation.
+   */
+  @SuppressWarnings("serial")
+  static ComponentInstanceSimpleDocumentAccessControlExtension getByInstanceId(final String instanceId) {
+    final SimpleCache cache = CacheServiceProvider.getRequestCacheService().getCache();
+    final String cacheKey = ComponentInstanceSimpleDocumentAccessControlExtension.class.getName() + "###" + instanceId;
+
+    final Mutable<ComponentInstanceSimpleDocumentAccessControlExtension> accessControlExtension =
+        Mutable.ofNullable(cache.get(cacheKey, ComponentInstanceSimpleDocumentAccessControlExtension.class));
+    if (accessControlExtension.isPresent()) {
+      return accessControlExtension.get();
+    }
+
+    try {
+      accessControlExtension.set(ServiceProvider
+          .getServiceByComponentInstanceAndNameSuffix(instanceId, Constants.NAME_SUFFIX));
+    } catch (IllegalStateException e) {
+      // Default implementation if none existing for the component
+      accessControlExtension.set(ServiceProvider.getService(
+          ComponentInstanceSimpleDocumentAccessControlExtension.class, new AnnotationLiteral<Base>() {}));
+    }
+    cache.put(cacheKey, accessControlExtension.get());
+    return accessControlExtension.get();
+  }
+
+  void beforeComputingAuthorizations(final AccessControlContext context);
+}
```

Comment:
I'm going to think about the use of cache to store a managed bean along the request processing scope: why not to put this way of caching directly in the ServiceProvider? By doing this, for example, the ServiceProvider first checks whether the asked bean is in the request cache, then in the application one if not (this invocation takes then also into account the singletons), and eventually invoke the underlying bean container implementation (here CDI) to get it.

Reply:
So, this optimization in finding programmatically a managed bean from CDI will be available for every client code.